### DISCUSSION TITLE
Some Examples

### DISCUSSION THREAD

**Discussion by StanisGit** [bbernhard/signal-cli-rest-api#379] [Q&A]
Could you provide more examples of working examples, such as registering a new user and (number) and sending attachments?

**Comment by bbernhard** [comment]
Have you seen the [examples](https://github.com/bbernhard/signal-cli-rest-api/blob/master/doc/EXAMPLES.md)?

**Comment by StanisGit** [comment]
Yes thank you. I saw examples, but something did not work out. I need to make a mailout to users from my program.
1. It turned out to send a simple message. Failed to send a large carriage return message (#13#10).
2. It turned out to create a group with users (users received an invitation to the group and joined on their own). I did not find how to add (invite) a new user to the old group.
3. It turned out to send a message to the group. But only I see messages in the group, other users do not see messages from signal-cli-rest-api. Users will only see messages sent through the Signal client. I thought that users need to register, but I was wrong.
{"error":"INFO AccountHelper - The Signal protocol expects that incoming messages are regularly received.\nFailed to send message due to untrusted identities\n"}
4. Tried to register another user. Register a number (with SMS verification) - does not work. Register a number (with voice verification) - doesn't work - "Captcha required for verification". It turned out to register (not the first time) the user through '{"captcha":"signal-hcaptcha......". Registration is required only for users on whose behalf messages are sent?
5. It turned out to verify the new user: curl -X POST -H "Content-Type: application/json" 'http://localhost:8080/v1/register/+123456789012/verify/123456'. But the Signal client asks to confirm the change of device: "This device is no longer registered. You probably registered your phone number with Signal on another device." Is it mandatory for users on whose behalf messages are sent and not mandatory for users who only want to receive messages? A more acceptable way with adding a device via http://localhost:8080/v1/qrcodelink?device_name=signal-api
6. It was not possible to send a picture to users, it is not clear where the file to be sent should be located.
curl -X POST -H "Content-Type: application/json" -d '{"message": "Test image", "base64_attachments": ["Signal/1.jpg"], "number": "+123456789012 ", "recipients": ["+123456789012"]}' 'http://localhost:8080/v2/send'
{"error":"illegal base64 data at input byte 8"}
I am using docker for bbernhard/signal-cli-rest-api:latest but want to send messages from my windows application

Examples that don't work:
 curl -X POST -H "Content-Type: application/json" -d '{"message": "Hello World!", "number": "+431212131491291", "recipients": ["group.ckRzaEd4VmRzNnJaASAEsasa", " +4912812812121"]}' 'http://127.0.0.1:8080/v2/send'
{"error":"Signal Messenger Groups and phone numbers cannot be specified together in one request! Please split them up into multiple REST API calls."}
